MelonLoader/Utils/MelonLogger.cs

```diff
@@ -267,19 +267,21 @@ internal static void PrintModName(Color meloncolor, Color authorcolor, string na
 
         internal static unsafe void PassLogMsg(ColorRGB msgColor, string msg, ColorRGB sectionColor, string section)
         {
+            char[] msgArray = msg.ToCharArray();
             if (section == null)
             {
-                fixed (char* pMsg = msg)
+                fixed (char* pMsg = msgArray)
                 {
                     BootstrapInterop.Library.LogMsg(&msgColor, pMsg, msg.Length, null, null, 0);
                 }
 
                 return;
             }
```

Comment:
While this is a fine solution for old mono, I don't agree having this on ALL runtimes. At this point, maybe let's just use marshaling to deal with the string for us?
There is no point in using `fixed` for a copy here, as the original intention was to pass the string without making any copies.

Reply:
done